chain/ethevents/events.go

```diff
@@ -117,6 +126,16 @@ func NewEthEvents(contractsAddresses []common.Address, signer *ethereum.SignKeys
 	if err != nil {
 		return nil, fmt.Errorf("cannot read token storage proof contract abi: %w", err)
 	}
+
+	secureAddrList := make([]common.Address, len(secureList))
```

Comment:
Better use a Map.
A Map is a hash table so for querying if a value exist is much faster than a slice.